pkg/logger/logger.go

```diff
@@ -55,13 +68,37 @@ func (l *logger) Info(msg string, fields ...Field) {
 }
 
 func (l *logger) Warn(msg string, fields ...Field) {
+	CaptureExceptions(fields)
 	l.underlyingLogger.Warn(msg, fields...)
 }
 
 func (l *logger) Error(msg string, fields ...Field) {
+	CaptureExceptions(fields)
 	l.underlyingLogger.Error(msg, fields...)
 }
 
 func (l *logger) Fatal(msg string, fields ...Field) {
+	CaptureExceptions(fields)
+	l.Sync()
+
 	l.underlyingLogger.Fatal(msg, fields...)
 }
+
+// Captures exceptions from fields and sends them to Sentry
+func CaptureExceptions(fields []Field) {
```

Comment:
I think this should be a private function
```suggestion
func captureExceptions(fields []Field) {
```